File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/implement/DomainConvertImp.java
package com.platdmit.simplecloudmanager.domain.converters.implement;

import com.platdmit.simplecloudmanager.data.api.models.ApiDomain;
import com.platdmit.simplecloudmanager.data.database.entity.DbDomain;
import com.platdmit.simplecloudmanager.domain.converters.DomainConverter;
import com.platdmit.simplecloudmanager.domain.models.Domain;

import java.util.ArrayList;
import java.util.List;

public class DomainConvertImp implements DomainConverter {

    @Override
    public DbDomain fromApiToDb(ApiDomain apiDomain) {
        return new DbDomain(apiDomain.getId(), apiDomain.getName(), apiDomain.getType(), apiDomain.getDeleteDate(), apiDomain.getIsDelegated());
    }

    @Override
    public Domain fromDbToDomain(DbDomain dbDomain) {
        return new Domain(dbDomain.getId(), dbDomain.getName(), dbDomain.getType(), dbDomain.getDeleteDate(), dbDomain.getIsDelegate());
    }

    @Override
    public Domain fromApiToDomain(ApiDomain apiDomain) {
        return new Domain(apiDomain.getId(), apiDomain.getName(), apiDomain.getType(), apiDomain.getDeleteDate(), apiDomain.getIsDelegated());
    }

    @Override
    public List<Domain> fromApiToDomainList(List<ApiDomain> apiList) {
        List<Domain> convertList = new ArrayList<>();
        for (ApiDomain apiDomain : apiList) {
            convertList.add(fromApiToDomain(apiDomain));
        }
        return convertList;
    }

    @Override
    public List<Domain> fromDbToDomainList(List<DbDomain> dbList) {
        List<Domain> convertList = new ArrayList<>();
        for (DbDomain dbDomain : dbList) {
            convertList.add(fromDbToDomain(dbDomain));
        }
        return convertList;
    }

    @Override
    public List<DbDomain> fromApiToDbList(List<ApiDomain> apiList) {
        List<DbDomain> convertList = new ArrayList<>();
        for (ApiDomain apiDomain : apiList) {
            convertList.add(fromApiToDb(apiDomain));
        }
        return convertList;
    }
}

File: domain/src/main/java/com/platdmit/domain/converters/implement/DomainConvertImp.kt
package com.platdmit.domain.converters.implement

import com.platdmit.data.api.models.ApiDomain
import com.platdmit.data.database.entity.DbDomain
import com.platdmit.domain.converters.DomainConverter
import com.platdmit.domain.models.Domain

class DomainConvertImp : DomainConverter {
    override fun fromApiToDb(apiDomain: ApiDomain): DbDomain {
        return DbDomain(apiDomain.id, apiDomain.name, apiDomain.type, apiDomain.deleteDate, apiDomain.isDelegated)
    }

    override fun fromDbToDomain(dbDomain: DbDomain): Domain {
        return Domain(dbDomain.id, dbDomain.name, dbDomain.type, dbDomain.deleteDate, dbDomain.isDelegate)
    }

    override fun fromApiToDomain(apiDomain: ApiDomain): Domain {
        return Domain(apiDomain.id, apiDomain.name, apiDomain.type, apiDomain.deleteDate, apiDomain.isDelegated)
    }

    override fun fromApiToDomainList(apiList: List<ApiDomain>): List<Domain> = apiList.map { fromApiToDomain(it) }

    override fun fromDbToDomainList(dbList: List<DbDomain>): List<Domain> = dbList.map { fromDbToDomain(it) }

    override fun fromApiToDbList(apiList: List<ApiDomain>): List<DbDomain> = apiList.map { fromApiToDb(it) }
}